### PR TITLE
Add Zig build system

### PR DESCRIPTION
**Pull request by gianni-rosato**
Also moved the C & h files into the `src` dir & updated the README

### CODE REVIEW

File: .gitignore
Comment (p7r0x7) [inline .gitignore:1]: Don't know that I support the addition of lines 1 to 53; they don't appear harmful, they just seem heavy-handed.

Comment (gianni-rosato) [reply .gitignore:1]: 🤷 its a template

**Comment by p7r0x7** [comment]
Hey, the zig build system here is compiling as c99, not c89; someone's gonna wanna change that.

**Comment by notcancername** [comment]
#4

**Pull request merged**